test/src/main/java/org/apache/accumulo/test/fate/accumulo/AccumuloStoreReadWriteIT.java

```diff
@@ -0,0 +1,124 @@
+import org.apache.accumulo.test.fate.zookeeper.FateIT.TestOperation;
+import org.junit.jupiter.api.Test;
+
+public class AccumuloStoreReadWriteIT extends AccumuloClusterHarness {
+
+  private static final NamespaceId NS = NamespaceId.of("testNameSpace");
+  private static final TableId TID = TableId.of("testTable");
+
+  @Override
+  protected Duration defaultTimeout() {
+    return Duration.ofMinutes(1);
+  }
+
+  @Test
+  public void testReadWrite() throws Exception {
+    final String table = getUniqueNames(1)[0];
+    try (ClientContext client =
+        (ClientContext) Accumulo.newClient().from(getClientProps()).build()) {
+      client.tableOperations().create(table);
+
+      AccumuloStore<Manager> store = new AccumuloStore<>(client, table);
+      // Verify no transactions
+      assertEquals(0, store.list().size());
+
+      // Create a new transaction and get the store for it
+      long tid = store.create();
+      FateTxStore<Manager> txStore = store.reserve(tid);
+      assertTrue(txStore.timeCreated() > 0);
+      assertEquals(1, store.list().size());
+
+      // Push a test FATE op and verify we can read it back
+      txStore.push(new TestOperation(NS, TID));
+      TestOperation op = (TestOperation) txStore.top();
+      assertNotNull(op);
+
+      // Test status
+      txStore.setStatus(TStatus.SUBMITTED);
+      assertEquals(TStatus.SUBMITTED, txStore.getStatus());
+
+      // Set a name to test setTransactionInfo()
+      txStore.setTransactionInfo(TxInfo.TX_NAME, "name");
+      assertEquals("name", txStore.getTransactionInfo(TxInfo.TX_NAME));
+
+      // Try setting a second test op to test getStack()
+      // when listing or popping TestOperation2 should be first
+      assertEquals(1, txStore.getStack().size());
+      txStore.push(new TestOperation2(NS, TID));
+      // test top returns TestOperation2
+      ReadOnlyRepo<Manager> top = txStore.top();
+      assertInstanceOf(TestOperation2.class, top);
+
+      // test get stack
+      List<ReadOnlyRepo<Manager>> ops = txStore.getStack();
+      assertEquals(2, ops.size());
+      assertInstanceOf(TestOperation2.class, ops.get(0));
+      assertEquals(TestOperation.class, ops.get(1).getClass());
+
+      // test pop, TestOperation should be left
+      txStore.pop();
+      ops = txStore.getStack();
+      assertEquals(1, ops.size());
+      assertEquals(TestOperation.class, ops.get(0).getClass());
+
+      // create second
+      FateTxStore<Manager> txStore2 = store.reserve(store.create());
+      assertEquals(2, store.list().size());
+
+      // test delete
+      txStore.delete();
+      assertEquals(1, store.list().size());
+      txStore2.delete();
+      assertEquals(0, store.list().size());
+    }
+  }
+
+  private static class TestOperation2 extends TestOperation {
+
+    private static final long serialVersionUID = 1L;
+
+    public TestOperation2(NamespaceId namespaceId, TableId tableId) {
+      super(namespaceId, tableId);
+    }
+  }
+
```

Comment:
I experimented with making Fate use this this new store by adding the following to the test and it worked.  

```java
public static class TestEnv {

  }

  public static class TestRepo implements Repo<TestEnv> {

    private final String data;

    TestRepo(String data) {
      this.data = data;
    }

    @Override
    public long isReady(long tid, TestEnv environment) throws Exception {
      System.out.println("Im ready "+data);
      return 0;
    }

    @Override
    public String getName() {
      return "TestRepo_"+data;
    }

    @Override
    public Repo<TestEnv> call(long tid, TestEnv environment) throws Exception {
      System.out.println("hello world : "+data);

      if(data.endsWith("_2")) {
        return null;
      }

      return new TestRepo(data+"_2");
    }

    @Override
    public void undo(long tid, TestEnv environment) throws Exception {

    }

    @Override
    public String getReturn() {
      return data+"_ret";
    }
  }

  @Test
  public void testFate() throws Exception {
    final String table = getUniqueNames(1)[0];
    try (ClientContext client =
                 (ClientContext) Accumulo.newClient().from(getClientProps()).build()) {
      client.tableOperations().create(table);

      AccumuloStore<TestEnv> store = new AccumuloStore<>(client, table);

      TestEnv testEnv = new TestEnv();

      Fate<TestEnv> fate = new Fate<>(testEnv, store, r->r+"", DefaultConfiguration.getInstance());

      long txid1 = fate.startTransaction();
      fate.seedTransaction("hw", txid1, new TestRepo("test_a"), false, "test accumulo store");

      fate.waitForCompletion(txid1);

      System.out.println("Return : "+fate.getReturn(txid1));

      fate.delete(txid1);

      fate.shutdown();
    }
  }
```

Saw the following output.

```
2023-12-14T17:20:10,104 [fate.Fate] INFO : Seeding FATE[21f670d0f198ad56] test accumulo store
Im ready test_a
hello world : test_a
Im ready test_a_2
hello world : test_a_2
Return : test_a_2_ret

```

Reply:
Nice, that's good that worked. My thinking for the initial testing of this with FATE itself was to work on updating/refactoring the FateIT so that we can run the tests against the ZK store and also the AccumuloStore but I need to look at it to see what changes the set up are needed. Maybe moves some of the test code into an abstract class or utilities or use parameterized testing, not sure yet.

Reply:
> Nice, that's good that worked. My thinking for the initial testing of this with FATE itself was to work on updating/refactoring the FateIT

I started experimenting with that locally, thought it might be a quick thing to try.  FateIT extends a super class that only starts ZK and not mini accumulo.  So it was a bigger change than I expected.